### ISSUE TITLE
ERROR ❌️ requires 2 classes, not 4

### ISSUE THREAD

**Issue by crapthings** [ultralytics/ultralytics#9135]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

train: /home/zznet/workspace/datasets/classify/train... found 3998 images in 2 classes ✅ 
val: /home/zznet/workspace/datasets/classify/val... found 1007 images in 4 classes: ERROR ❌️ requires 2 classes, not 4

<img width="1596" alt="image" src="https://github.com/ultralytics/ultralytics/assets/1147704/41a0ad28-6aca-42d5-9c06-d16b1ac4c8c8">

root

<img width="442" alt="image" src="https://github.com/ultralytics/ultralytics/assets/1147704/63bc5037-9958-495d-9a36-b34e5196744a">

train

<img width="308" alt="image" src="https://github.com/ultralytics/ultralytics/assets/1147704/9708267b-1bc8-4040-ae81-ac6ed6b94283">

val

<img width="330" alt="image" src="https://github.com/ultralytics/ultralytics/assets/1147704/88be3e5e-2ec7-49d3-af37-912a34d30c5a">

### Environment

ubuntu server
Python 3.10.12
ultralytics               8.1.29

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@crapthings hello! It looks like there's a mismatch in the number of classes between your training and validation datasets. Your training set found images belonging to 2 classes, but the validation set contains images from 4 classes. To resolve this, make sure both datasets contain the same number of classes and that the class names in your `data.yaml` file accurately reflect this. Here's a quick example of how your `data.yaml` classes section should look if you have, say, 2 classes:

```yaml
nc: 2  # number of classes
names: ['class0', 'class1']  # class names
```

Make any necessary adjustments to ensure both your training and validation sets are consistent with each other in terms of class labels. This should fix the error you're encountering. Hope this helps! 😊

**Comment by haimat** [comment]
> @crapthings hello! It looks like there's a mismatch in the number of classes between your training and validation datasets. Your training set found images belonging to 2 classes, but the validation set contains images from 4 classes. To resolve this, make sure both datasets contain the same number of classes and that the class names in your `data.yaml` file accurately reflect this. Here's a quick example of how your `data.yaml` classes section should look if you have, say, 2 classes:
> 
> ```yaml
> nc: 2  # number of classes
> names: ['class0', 'class1']  # class names
> ```
> 
> Make any necessary adjustments to ensure both your training and validation sets are consistent with each other in terms of class labels. This should fix the error you're encountering. Hope this helps! 😊

I have exactly the same problem, as described in https://github.com/ultralytics/ultralytics/issues/8837
But there you also say that a yaml config file is not required for YOLO classification.
And passing the `nc` argument directly to `train()` does also not work.
So how else can we specify the number of classes to train?
